Form/JQuery/Type/Select2Type.php

```diff
@@ -89,13 +90,16 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
      */
     public function getParent()
     {
-        return $this->widget;
+        if (class_exists('Symfony\Component\Form\Extension\Core\Type\\' . ucfirst($this->widget) . 'Type')) {
```

Comment:
I use this condition to check whether the class name exist or not. Since from Symfony 2.8 onward, `getParent()` return the fully-qualified class name. Simply returning the widget name will trigger deprecation. Mostly the name type fall under this folder `Symfony\Component\Form\Extension\Core\Type`